### PR TITLE
addons: find and use the best matching translation for summary/description/disclaimer

### PR DESCRIPTION
**Pull request by Montellese**
@alanwww1 made me aware of the fact that when a user has `Portuguese (Brazilian)` set as his GUI language and an addon has a summary/description/disclaimer for both `pt` as well as `pt_BR` it depends on the order of the tags in the XML whether we choose the translated strings for `pt` or `pt_BR` but it's usually the one for `pt`.

I have added a helper method to `CLocale` which takes a list of locales and tries to find the one that matches the current locale best. If there's an exact match, that locale will be taken. Otherwise we look for the locale that is closest to the current locale.
`CAddonMgr` makes use of that function by collecting all translated versions of summary/description/disclaimer that at least partly match in locale and then tries to find the best match and returns the translated value of the best matching locale.

### CODE REVIEW

File: xbmc/utils/Locale.cpp [hunk 1]
```diff
@@ -233,3 +252,30 @@ void CLocale::Initialize()
     StringUtils::ToUpper(m_territory);
   }
 }
+
+int CLocale::GetMatchCount(const std::string& locale) const
```
Comment (ace20022) [inline xbmc/utils/Locale.cpp]: It's quite early at the morning  so I can be wrong ;) I guess you calculate a score or rank here. If so I would rename the method.

File: xbmc/utils/Locale.cpp [hunk 2]
```diff
@@ -233,3 +252,30 @@ void CLocale::Initialize()
...
+{
+  CLocale other = FromString(locale);
+
+  if (!m_valid && !other.m_valid)
```
Comment (ace20022) [inline xbmc/utils/Locale.cpp]: This check is superflous, isn't it?

**Comment by Montellese** [comment]
@ace20022: Thanks for the review. Adressed both of your comments.

**Comment by alanwww1** [comment]
@Montellese
I have tested this PR. Compiled ok, and working great.
I tested with pt_BR and  even with fr_CA. Handled them perfectly.

**Comment by Montellese** [comment]
@alanwww1: Cool thanks.
@ace20022: Anything else?
jenkins build this please

**Comment by ace20022** [comment]
"bestMatchRank" doesn't get updated if I see it right. For the rest I would need a debugger ;) I have no time for tests till sunday, but I trust you that it's correct.

**Comment by mkortstiege** [comment]
Jep, `bestMatchRank` looks indeed a bit fishy.

**Comment by Montellese** [comment]
Hm must have accidentally removed it during the rebase. Thanks for the hint.

**Comment by Montellese** [comment]
Updated. jenkins build this please

**Pull request merged**